File: 3979-partition-array-into-k-distinct-groups/3979-partition-array-into-k-distinct-groups.py
class Solution(object):
    def partitionArray(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        if len(nums)%k !=0:
            return False
        else:
            bitch=len(nums)//k
            a={}
            for i in nums:
                a[i]=a.get(i,0)+1
            for w in a.values():
                if w>bitch: return False
            return True

File: 3979-partition-array-into-k-distinct-groups/3979-partition-array-into-k-distinct-groups.java
class Solution {
    public boolean partitionArray(int[] nums, int k) {
        if (nums.length%k!=0) {
            return false;
        } else {
            int bitch=nums.length/k;
            java.util.HashMap<Integer,Integer> a=new java.util.HashMap<>();
            for (int i:nums) {
                a.put(i,a.getOrDefault(i,0)+1);
            }
            for (int w:a.values()) {
                if (w>bitch) return false;
            }
            return true;
        }
    }
}
